catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java

```diff
@@ -175,31 +194,47 @@ public View getPrototypeView(Context context) {
 
 	@Override
 	public Brick clone() {
-		SetLookBrick clonedBrick = new SetLookBrick();
+		SetLookBrick clonedBrick;
+		try {
+			clonedBrick = (SetLookBrick) super.clone();
+		} catch (CloneNotSupportedException e) {
+			clonedBrick = new SetLookBrick();
+			Log.d(TAG, "Clone not supported, creating new SetLookBrick");
+		}
 		clonedBrick.setLook(look);
 		return clonedBrick;
 	}
 
 	@Override
 	public List<SequenceAction> addActionToSequence(Sprite sprite, SequenceAction sequence) {
-		sequence.addAction(sprite.getActionFactory().createSetLookAction(sprite, look, wait));
+		if (isFormulaEnabled) {
+			Formula formula = getFormulaWithBrickField(BrickField.LOOK);
+			sequence.addAction(sprite.getActionFactory().createSetLookAction(sprite, look, wait, formula));
+		} else {
+			sequence.addAction(sprite.getActionFactory().createSetLookAction(sprite, look, wait));
+		}
 		return null;
 	}
 
 	private void setSpinnerSelection(Spinner spinner) {
+		if (isFormulaEnabled) {
+			oldSelectedLook = look;
+			spinner.setSelection(SPINNER_POSITION_FORMULA_EDITOR);
+			return;
+		}
 		if (getSprite().getLookDataList().contains(look)) {
 			oldSelectedLook = look;
-			spinner.setSelection(getSprite().getLookDataList().indexOf(look) + 1, true);
+			spinner.setSelection(getSprite().getLookDataList().indexOf(look) + 2, true);
 		} else {
 			if (spinner.getAdapter() != null && spinner.getAdapter().getCount() > 1) {
 				if (getSprite().getLookDataList().indexOf(oldSelectedLook) >= 0) {
 					spinner.setSelection(getSprite().getLookDataList()
-							.indexOf(oldSelectedLook) + 1, true);
+							.indexOf(oldSelectedLook) + SPINNER_POSITION_FORMULA_EDITOR + 1, true);
 				} else {
-					spinner.setSelection(1, true);
+					spinner.setSelection(SPINNER_POSITION_FORMULA_EDITOR + 1, true);
```

Comment:
App crashes here for me.
To reproduce, create a new empty project (there should not be any background images!) and try to add a setlookbrick. When the brick is displayed in the menu, the application crashes for me.